tests/integration/Cake.Core/Scripting/SystemCollections.cake

```diff
@@ -0,0 +1,10 @@
+Task("Cake.Core.Scripting.SystemCollections.ToDictionary")
+    .Does(() =>
+{
+    var dictionary = Array.Empty<int>().ToDictionary(key => key, value=>value);
```

Comment:
I wasn't thinking when I gave the sample, but this could be more direct:

```diff
-    var dictionary = Array.Empty<int>().ToDictionary(key => key, value=>value);
+    Array.Empty<int>().ToDictionary(key => key);
```